### PR TITLE
[GNA] Added support permute layer

### PR DESCRIPTION
**Pull request by admitriev-gna**
Added support permute layer
Added tests

### CODE REVIEW

File: inference-engine/src/gna_plugin/layers/gna_layer_info.hpp [hunk 1]
```diff
@@ -194,7 +195,43 @@ class LayerInfo {
     bool isPermute() const noexcept {
         return isOfType("permute");
     }
-    bool isPooling() const noexcept {
+        // @brief this not only mathematically trivial, has some WA for kaldi case
+        bool isTrivialPermute() {
```
Comment (pavel-rodionov) [inline inference-engine/src/gna_plugin/layers/gna_layer_info.hpp:199]: It looks like formatting here is not correct. @admitriev-gna Could you please double check?

File: inference-engine/src/gna_plugin/gna_graph_compiler.cpp [hunk 2]
```diff
@@ -1412,30 +1413,76 @@ case name:\
...
+
+    if (squeezedInputOrder.size() > 2) {
+        THROW_GNA_EXCEPTION << "unsupported permute (requested transpose is not 2D)";
+    }
 
-    string dimMessage;
-    if (layerOrder == vector<int>({0, 3, 2, 1})) {
-        return;  // supported case
+    if (std::min(squeezedInputOrder[0], squeezedInputOrder[1]) > 8) {
+        THROW_GNA_EXCEPTION << "unsupported permute (minor dimension="
```
Comment (pavel-rodionov) [inline inference-engine/src/gna_plugin/gna_graph_compiler.cpp:1443]: Please change it to `THROW_GNA_LAYER_EXCEPTION` so layer name will be printed to log

File: inference-engine/src/gna_plugin/backend/am_intel_dnn.cpp [hunk 3]
```diff
@@ -311,6 +311,65 @@ void GNAPluginNS::backend::AMIntelDNN::InitPiecewiseLinearComponentPrivate(intel
...
+    comp.num_rows_in = num_rows_in;
+    comp.num_columns_in = num_columns_in;
+    comp.num_rows_out = num_columns_in;
+    comp.num_columns_out = num_rows_in;
+    comp.num_bytes_per_input = num_bytes_per_input;
+    comp.num_bytes_per_output = num_bytes_per_output;
+    comp.operation = kDnnDeinterleaveOp;
+    comp.macro_operation = kDnnMacroOpNone;
+    comp.orientation_in = kDnnNonInterleavedOrientation;
+    comp.orientation_out = kDnnInterleavedOrientation;
```
Comment (pavel-rodionov) [inline inference-engine/src/gna_plugin/backend/am_intel_dnn.cpp:362]: It looks like both `Deinterleave` and `Interleave` returns primitive with `kDnnInterleavedOrientation` 
Could you please double check this?

File: inference-engine/src/gna_plugin/layers/gna_permute.hpp [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+
+#pragma once
+#include <vector>
+#include <list>
+#include <utility>
+#include "gna_plugin_log.hpp"
+
+namespace GNAPluginNS {
+template <class T>
+class PermuteSequence {
```
Comment (esmirno) [inline inference-engine/src/gna_plugin/layers/gna_permute.hpp:14]: please restore unit tests for permute sequence as i remember there lots of cases tested

**Comment by dorloff** [comment]
Some VPU test failed - probably needs rebasing

**Comment by esmirno** [comment]
lgtm - permute sequence tests are passed, thanks

**Pull request merged**